### ISSUE TITLE
Feature Request: Night Mode

### ISSUE THREAD

**Issue by Bouncey** [freeCodeCamp/learn#184]
It's about time we had a night mode on this thing.

There is a prop on the user object that holds the current users theme selection. That prop is `theme`.

`theme` is a string that can be either `'default'` or `'night'`.

## User Stories 
- The learn UI will reflect my theme choice. 
- A selection of `default` will give me the current theme on `learn`
- A selection of `night` will give me a night mode ui similar to the night mode at `www.freecodecamp.org` (The CSS should still exist in that codebase, freecodecamp/freecodecamp/client/main.less, if I remember correctly)

**Comment by renojvarghese** [comment]
I'd like to take this one.

**Comment by Bouncey** [comment]
Awesome! Thanks @renojvarghese

**Comment by renojvarghese** [comment]
Hey. I have a couple questions to complete this feature.

1. I found this file: https://github.com/freeCodeCamp/learn/blob/master/src/html.js It takes in props for bodyAttributes. I can add a night class to the body. Where is this component rendered? I couldn't find a reference to it.
2. How can I access the user object? I believe it would be in the redux store.

**Comment by Bouncey** [comment]
@renojvarghese `html.js` is used by `gatsby`, we do not reference it directly. 

instead of attaching a class to the body, you could add a class to the [`.app-wrapper`](https://github.com/freeCodeCamp/learn/blob/069f551ffb9927886743412bfffd5b2870d350a0/src/layouts/index.js#L94)

`.app-warpper` is where we render everything in learn.

**Comment by renojvarghese** [comment]
The background color doesn't extend to the bottom of the page when `.app-wrapper` has the `.night` class. 

Here is a screenshot of `.app-wrapper` with `.night`
<img width="1207" alt="screen shot 2018-06-20 at 8 46 39 pm" src="https://user-images.githubusercontent.com/24216993/41691887-1a4b2ea6-74cb-11e8-93b0-c2c8984a0762.png">


Here is a screenshot of body with `.night`:
<img width="1215" alt="screen shot 2018-06-20 at 8 45 16 pm" src="https://user-images.githubusercontent.com/24216993/41691848-e5a342f6-74ca-11e8-91ed-40f2b1bef0c4.png">

Also that green is a bit hard to read on the gray background. May I change the green to the one in https://github.com/freeCodeCamp/freeCodeCamp/pull/17514

**Comment by Bouncey** [comment]
You could look in to why `.app-wrapper` is not at 100% height, if that can be fixed without affecting other views then that is great.

Re the green, feel free to use the same colour the rest of the map text is in, looks pretty readable to me.

**Comment by QuincyLarson** [comment]
@renojvarghese Thanks for helping with the night mode on Learn. People ask me about this every day, and I'm thrilled we're so close to getting it live. How is everything going with this? Is there anything I can do to help?

**Comment by renojvarghese** [comment]
Yea. Happy to help. So far, I have all the styles in, but I can't sign in to test if if night mode is turned on by the user's settings. I tried to make an account on FCC locally, but I run into an error. I made a post in the contributor Gitter, and it seems I am blocked by this issue: https://github.com/freeCodeCamp/freeCodeCamp/issues/17382

Is there a way to see a mock user object or use another workaround?

**Comment by QuincyLarson** [comment]
@renojvarghese If you have all the styles, could you go ahead and open a pull request? @raisedadead can QA locally and we can fix this sooner while we're still working on the fix for freeCodeCamp/freeCodeCamp#17382

Thanks for your help with this. People ask me several times a day for this feature!

**Comment by renojvarghese** [comment]
@Bouncey  @QuincyLarson Pull request is in: https://github.com/freeCodeCamp/learn/pull/202

Here are a few screenshots of what I came up with:

<img width="1213" alt="screen shot 2018-06-28 at 8 31 01 pm" src="https://user-images.githubusercontent.com/24216993/42067202-08046d90-7b13-11e8-8bf8-b294cf3c752b.png">
<img width="1211" alt="screen shot 2018-06-28 at 8 34 41 pm" src="https://user-images.githubusercontent.com/24216993/42067204-09de91f4-7b13-11e8-9fdf-2a914748a48a.png">

Let me know what you think.

**Comment by johnkennedy9147** [comment]
Couple of issues with night mode on Learn:

Cursor hard to see in editor. Black on dark grey.

No visual indication at all of selected text - in below screenshot I have selected `function minusSeven`
![image](https://user-images.githubusercontent.com/26827682/42100963-9b68e452-7bb9-11e8-9f28-48fae3be047f.png)

I think we should revert this if not able to fix really quickly.
@Bouncey @raisedadead @QuincyLarson

**Comment by Bouncey** [comment]
Hi @johnkennedy9147, could you raise another issue for it, it should be a quick css fix

**Comment by johnkennedy9147** [comment]
see #204
